src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -1305,8 +1306,57 @@ Error maybeCopyWebsiteAsset(const json::JsonRpcRequest& request,
    return Success();
 }
 
-
-
+Error executeAlternateEngineChunk(const json::JsonRpcRequest& request,
+                                  json::JsonRpcResponse* pResponse)
+{
+   std::string docId, chunkId, engine, code;
+   Error error = json::readParams(request.params,
+                                  &docId,
+                                  &chunkId,
+                                  &engine,
+                                  &code);
+   if (error)
+   {
+      LOG_ERROR(error);
+      return error;
+   }
+   
+   if (string_utils::toLower(engine) == "python")
+   {
+      // write code to temporary file
+      FilePath codePath = module_context::tempFile("python", ".py");
+      error = core::writeStringToFile(codePath, code);
+      if (error)
+      {
+         LOG_ERROR(error);
+         return error;
+      }
+      
+      // create command to execute
+      core::shell_utils::ShellCommand command("python");
+      command << codePath.absolutePathNative();
+      
+      // create process
+      boost::shared_ptr<ExecuteChunkOperation> operation =
```

Comment:
Right; I'm thinking these process objects will live in a map / registry of some kind (so that you can access a process,  if any,  by a combination of document + chunk id)